### PR TITLE
Update `getstationvariabletable` to distinguish climo variables correctly

### LINKED ISSUE #184: Function `getstationvariabletable` uses incorrect method to discriminate climatological variables
For details on the required update, see [Redmine ticket](https://redmine.pcic.uvic.ca/issues/1974) in which this was done ad-hoc.

### PR DESCRIPTION
**Pull request by rod-glover**
Resolves #184 

In addition to fixing `getstationvariabletable`, this PR also moves function definitions out to module `pycds.orm.functions`, something I have been meaning to do for a while. Reviewers should feel free to ignore this.

### CODE REVIEW

File: tests/alembic_migrations/versions/v_efde19ea4f52_fix_function_getstationvariabletable/conftest.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        db_setup=db_setup,
+        revision="efde19ea4f52",
+    )
+
+    yield engine, script
+
+    engine.dispose()
+
+
+@pytest.fixture(scope="function")
```
Comment (Nospamas) [inline tests/alembic_migrations/versions/v_efde19ea4f52_fix_function_getstationvariabletable/conftest.py:21]: It doesn't look like this gets used.

File: tests/alembic_migrations/versions/v_efde19ea4f52_fix_function_getstationvariabletable/test_smoke.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    # Set up database to version efde19ea4f52
+    engine, script = prepared_schema_from_migrations_left
+
+    # Run downgrade migration
+    command.downgrade(alembic_config_left, "-1")
+
+    # Check that function is still there (old version)
+    names = get_schema_item_names(engine, "routines", schema_name=schema_name)
+    assert "getstationvariabletable" in names
```
Comment (Nospamas) [inline tests/alembic_migrations/versions/v_efde19ea4f52_fix_function_getstationvariabletable/test_smoke.py:42]: Is there any call for testing the actual data function change? or is that covered by pre-existing tests?

**Review by Nospamas — CHANGES_REQUESTED** [review]
Mostly points for quick discussion before approval.

File: tests/alembic_migrations/versions/v_efde19ea4f52_fix_function_getstationvariabletable/test_smoke.py (hunk 2 above)
Comment (rod-glover) [reply tests/alembic_migrations/versions/v_efde19ea4f52_fix_function_getstationvariabletable/test_smoke.py:42]: There are no tests for the behaviour, which is to say input-output relations of this function, or any others. There never have been; they were manually entered in pre-PyCDS times. More recently, they were brought under PyCDS management, with minimal testing (only that they can be invoked without error). This is not a good state of affairs. See #198 .

File: pycds/alembic/versions/efde19ea4f52_fix_function_getstationvariabletable.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+# revision identifiers, used by Alembic.
+revision = "efde19ea4f52"
+down_revision = "6cb393f711c3"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    op.set_role(get_su_role_name())
```
[the remark this answers is not in the document]
Comment (rod-glover) [reply pycds/alembic/versions/efde19ea4f52_fix_function_getstationvariabletable.py:26]: From another migration (... this needs to be made more visible):

```
    # In order to create stored procedures using untrusted languages (we use
    # `plpython3u`), the user needs superuser privileges. This is achieved by
    # temporarily setting the role to a superuser role name that is externally
    # granted to the user only for the period when database migrations are
    # performed.
```

File: tests/alembic_migrations/versions/v_efde19ea4f52_fix_function_getstationvariabletable/conftest.py (hunk 1 above)
Comment (rod-glover) [reply tests/alembic_migrations/versions/v_efde19ea4f52_fix_function_getstationvariabletable/conftest.py:21]: No, it doesn't and worse this pattern is all over the place in the migration smoke tests. I'll put this off to a different issue #201 .

File: pycds/alembic/versions/efde19ea4f52_fix_function_getstationvariabletable.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (rod-glover) [reply pycds/alembic/versions/efde19ea4f52_fix_function_getstationvariabletable.py:26]: Added to documentation

**Review by Nospamas — APPROVED** [review]
:+1:

**Pull request merged**